Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/Concepts.java

```diff
@@ -201,35 +274,77 @@ private JsonValidationResult checkJsonObject(JSONObject jsonObject) {
                     false);
         }
 
-        if (jsonObject.get(JsonFields.CONCEPT_LIST) == null) {
+        if (jsonObject.get(JsonFields.DESCRIPTION) == null) {
             return new JsonValidationResult(
-                    "Error parsing request: please provide a concept list for the concept ('conceptlist' attribute).",
+                    "Error parsing request: please provide a description for the concept ('description' attribute).",
                     jsonObject, false);
         }
 
-        String pos = jsonObject.get(JsonFields.POS).toString();
-        if (!POS.posValues.contains(pos)) {
-            logger.error("Error creating concept from REST call. " + pos + " does not exist.");
-            return new JsonValidationResult("POS '" + pos + "' does not exist.", jsonObject, false);
+        // Validation to check if wordnet ids are seperated by comma
+        if (jsonObject.get(JsonFields.WORDNET_ID) != null) {
+            String wordnetIds = jsonObject.get(JsonFields.WORDNET_ID).toString();
+            List<String> wordnetIdsList = Arrays
+                    .asList(wordnetIds.split("\\s*" + Constants.CONCEPT_SEPARATOR + "\\s*"));
+            for (String wordNetId : wordnetIdsList) {
+                try{ 
+                    WordID.parseWordID(wordNetId);
+                } catch (IllegalArgumentException ex) {
+                    return new JsonValidationResult(
+                            "Error parsing request: please provide a valid list of wordnet ids seperated by commas. Wordnet id "
+                                    + wordNetId + " doesn't exist.",
+                            jsonObject, false);
+                }
+            }
+
+            // Check if there is an existing wrapper concept if so throw error
+            for (String wordNetId : wordnetIdsList) {
+                if (conceptManager.getConceptWrappedEntryByWordNetId(wordNetId) != null) {
+                    // This wordnet id is already wrapped. Throw error
+                    return new JsonValidationResult(
+                            "Error parsing request: please provide a valid wordnet id that doesn't have a wrapped concept.",
```

Comment:
should say:
"...: the WordNet concept you are trying to wrap is already wrapped."